### DISCUSSION TITLE
Generate Typescript model from Supabase Swagger

### DISCUSSION THREAD

**Discussion by emeraldjava** [supabase/supabase#1366] [Questions] [answered]
Hi,

I appreciate the supabase client uses the postgres API and the meta-data from the database to build queries. Building on top of the suapbase-js client in a node UI application, I'd like to have typescript model interface/classes generated for my DB schema. 

There are existing tools like [swagger-angular-generator](https://www.npmjs.com/package/swagger-angular-generator) which achieve this.

 Is there a protected URL which exposes the swagger OpenAPI specification for my database which i could point this tool at?
Thanks,
emeraldjava

**Comment by easy-street** [comment, marked as answer]
sounds like the `/rest/v1/` (Introspection) endpoint may be what you're looking for?:

```
curl --request GET \
  --url <SUPABASE_URL>/rest/v1/ \
  --header 'apikey: <SUPABASE_ANON_KEY>'
```

**Reply by emeraldjava** [reply]
Yeah - that's the endpoint I required. Thanks

**Comment by emeraldjava** [comment]
Using the endpoint mentioned above I then use the [openapi-typescript](https://www.npmjs.com/package/openapi-typescript) tools to generate a typescript file with the interface defitions for my DB model.

`npx openapi-typescript <SUPABASE_URL>/rest/v1/?apikey=<SUPABASE_ANON_KEY> --output types/supabase.ts`
